### ISSUE TITLE
Document mount order

### ISSUE THREAD

**Issue by flx42** [lxc/lxc#2932]
Mostly a nice to have, but I got confused for a while on why our `mount` hook ("a hook to be run in the container's namespace after mounting has been done, but before the pivot_root") was called before the `lxc.mount.entry` but after the `lxc.mount.auto`. 

From the code:
https://github.com/lxc/lxc/blob/9810d1916eacea64c838681d3045cac60e7f00c2/src/lxc/conf.c#L3537-L3614
The order is:
* `lxc.mount.auto = proc` and `lxc.mount.auto = sys`
* `lxc.mount.fstab`
* `lxc.mount.auto = cgroup`
* `lxc.hook.mount`
* `lxc.autodev`
* `lxc.mount.entry`

By the way, was there a reason to **not** have `lxc.mount.entry` immediately after `lxc.mount.fstab`?

**Comment by brauner** [comment]
> By the way, was there a reason to not have lxc.mount.entry immediately after lxc.mount.fstab?

Not that I know of.

You want the mount order documented in `lxc.container.conf` or in the code?

**Comment by flx42** [comment]
> By the way, was there a reason to not have lxc.mount.entry immediately after lxc.mount.fstab?

Should we try to fix it or is too late and part of the implicit API?

> You want the mount order documented in lxc.container.conf or in the code?

In `lxc.container.conf`, the code is easy to follow once you find the right place.

**Comment by brauner** [comment]
Yeah, I think we can send a patch, clean this up and then document it. :)

**Comment by brauner** [comment]
Care to send one, @flx42?

**Comment by flx42** [comment]
Was out of the office for a few days, I'll try to do it this week.

**Comment by flx42** [comment]
Let's see if changing the mount order breaks anything before tackling documentation.

**Comment by tardyp** [comment]
> By the way, was there a reason to not have lxc.mount.entry immediately after lxc.mount.fstab?

A use case is to bind mount additional stuff like /dev/bus/usb

This is documented to be working here:
https://medium.com/@konpat/usb-passthrough-to-an-lxc-proxmox-15482674f11d

does not work for me, but I am in lxc 3.03 so before this patch..
